src/services/download/sentry.rs

```diff
@@ -204,20 +208,18 @@ pub fn list_files(
 
         url
     };
-
     let query = SearchQuery {
         index_url,
         token: source.token.clone(),
     };
 
-    let entries = perform_search(query).map(clone!(source, |entries| {
-        entries
-            .into_iter()
-            .map(move |api_response| {
-                SourceFileId::Sentry(source.clone(), SentryFileId::new(api_response.id))
-            })
-            .collect()
-    }));
-
-    Box::new(entries)
+    let search = cached_sentry_search(query);
+    let entries = future_utils::time_task("downloads.sentry.index", search)
+        .await?
+        .iter()
```

Comment:
Can you use `into_iter` here? That should avoid the clone of `search_result.id`.
_nit_: You can also break this up in two statements. One that awaits, and then the next one that maps.

Reply:
I wonder why clippy can't warn for `.into_iter()`